DEVELOPER_GUIDE.md

```diff
@@ -2,74 +2,71 @@
- - Here we define 2 generic functions, `lift2` which corresponds always to the applicative instance, so normally in non-collections it will correspond to the non-generic `map2` but since collections already have (or at least it's expected to) a `map2` acting pairwise, in those cases only we provide a `lift2` extension and that's what's used for its generic counterpart.
- 
- - The other generic function we provide is `zip` which is available mainly for collection like types and although they match the non-generic name, note that the behavior is not exactly the same, because F# core throws errors for list and arrays when the number of elements are different. So we could say that in fact it matches `.zipShortest` when defined, otherwise it matches `.zip`.
+- For collections like types, in F# core it's normal to find `zip` / `map2` functions, which acts pairwise. But another possible implementation is the applicative zip which works cross product.
+- Here we define 2 generic functions, `lift2` which corresponds always to the applicative instance, so normally in non-collections it will correspond to the non-generic `map2` but since collections already have (or at least it's expected to) a `map2` acting pairwise, in those cases only F#+ provides a `lift2` extension and that's what's used for its generic counterpart.
+- The other provided generic function is `zip` which is available mainly for collection like types and although they match the non-generic name, note that the behaviour is not exactly the same, because F# core throws errors for list and arrays when the number of elements are different. In that sense, in fact, it matches `.zipShortest` when defined, otherwise it matches `.zip`.
 
-So these zip related functions are a good example of how we do our best to fix FSharp.Core inconsistencies in a non-intrusive way.
+So these zip related functions are a good example of how F#+ does its best to fix FSharp.Core inconsistencies in a non-intrusive way.
 
 
 ### Abstractions
 
-The abstractions represented in the Control namespace of this library are a set of types that in fact represents generic functions, we call these types and we'll refer to as "Invokables".
+The abstractions represented in the Control namespace of this library are a set of types that in fact represents generic functions, these types are referred to as `Invokables`.
 
-These Invokables are organized in such a way that resemble static interfaces, typeclasses, concepts or traits in other languages where they have a type system construct enabling higher kinds and ad-hoc polymorphism.
+These `Invokables` are organized in such a way that resemble `static interfaces`, `typeclasses`, `concepts` or `traits` in other languages where they have a type system construct enabling higher kinds and ad-hoc polymorphism.
```

Comment:
> Java's `static interface`, Haskell's `class`, C++'s `concept`, or Rust's `trait`

I think code quotes are fine here since they refer to the actual syntax in those languages.

Reply:
I would rather avoid mentioning any particular language and leave it more loose.
And, as now .Net has `static interfaces` it might change in the future.